gate_block/models.py

```diff
@@ -32,17 +32,17 @@ def allow_redo(self):
     def unlocked(self, user, section):
         return GateSubmission.objects.filter(
             gateblock_id=self.id,
-            gate_user_id=user.id).count() > 0
+            gate_user_id=user.id).exists()
 
     def status(self, gate_section, user, hierarchy, uloc, pageblocks=None):
         """
         Takes self.pageblock().section, a User, Hierarchy, UserLocation,
         and optionally, a pageblock set.
         """
-        pageblocks = self._get_pageblocks(pageblocks, gate_section)
 
-        ss_exists = SectionSubmission.objects.filter(
-            user=user, section=gate_section).exists()
+        # @todo: could these be passed in?
+        pageblocks = self._get_pageblocks(pageblocks, gate_section)
+        ss_exists = gate_section.section_submited.filter(user=user).exists()
```

Comment:
@todo - [submited is mispelled](https://github.com/ccnmtl/uelc/blob/master/gate_block/models.py#L111). I'll address in a separate pr.